### PR TITLE
Fix `DeclareLaunchArgument` xml parsing and constructor

### PR DESCRIPTION
**Pull request by ivanpauno**
Fixes https://github.com/ros2/launch/issues/524 and another bug I found while writing the added test case.

### CODE REVIEW

File: launch_xml/test/launch_xml/test_arg.py [hunk 1]
```diff
@@ -38,6 +38,27 @@ def test_arg():
...
+            </arg>
+        </launch>
+        """
+    xml_file = textwrap.dedent(xml_file)
+    root_entity, parser = Parser.load(io.StringIO(xml_file))
+    ld = parser.parse_description(root_entity)
+    arg = ld.entities[0]
+    assert 'my_arg' == arg.name
+    assert 'asd' == ''.join([x.perform(None) for x in arg.default_value])
+    assert "something Valid choices are: ['asd', 'bsd']" == arg.description
```
Comment (hidmic) [inline launch_xml/test/launch_xml/test_arg.py:58]: @ivanpauno nit^N: I know this is pre-existing code, but could we ensure there's a period between the description and the list of choices:
```suggestion
    assert "something. Valid choices are: ['asd', 'bsd']" == arg.description
```
?

**Review by hidmic — APPROVED** [review]
LGTM pending green CI

File: launch_xml/test/launch_xml/test_arg.py (hunk 1 above)
Comment (ivanpauno) [reply launch_xml/test/launch_xml/test_arg.py:58]: the test doesn't path with that change, I would need to change the source code for that

Comment (hidmic) [reply launch_xml/test/launch_xml/test_arg.py:57]: Right, that's what I meant by ensuring there's a period :sweat_smile: 

Sorry for the noisy comms.

Comment (ivanpauno) [reply launch_xml/test/launch_xml/test_arg.py]: > Right, that's what I meant by ensuring there's a period sweat_smile

Sorry, I didn't read the first part of your original comment, I only saw the suggestion.
Fixed in https://github.com/ros2/launch/pull/529/commits/5d860a8693aed53ee80b0ec63cb45a8ffbd005ec

**Comment by ivanpauno** [comment]
CI:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/15192/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/9912/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/12872/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/15373/)

**Review by hidmic — APPROVED** [review]

**Pull request merged**